engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ScriptAssertContext.java

```diff
@@ -54,14 +58,16 @@ public boolean evaluateScriptAssertExpression(Map<String, Object> bindings) {
 		return handleResult( result );
 	}
 
-	private ScriptEvaluator getScriptEvaluator(String languageName) {
-		try {
-			ScriptEvaluatorFactory evaluatorFactory = ScriptEvaluatorFactory.getInstance();
-			return evaluatorFactory.getScriptEvaluatorByLanguageName( languageName );
-		}
-		catch (ScriptException e) {
-			throw log.getCreationOfScriptExecutorFailedException( languageName, e );
+	private synchronized ScriptEvaluator getScriptEvaluator( ConstraintValidatorContext context) {
```

Comment:
So if we move that to the Validator, please don't use a plain `synchronized` method but take a look at http://www.javamex.com/tutorials/double_checked_locking_fixing.shtml , especially the last paragraph: Use DCL plus volatile (except the synchronized should be on (this) not on the class).